sdk/core/azure-core/CMakeLists.txt

```diff
@@ -23,13 +23,30 @@ find_package(Threads REQUIRED)
 
 # min version for `CURLSSLOPT_NO_REVOKE`
 # https://curl.haxx.se/libcurl/c/CURLOPT_SSL_OPTIONS.html
-set(CURL_MIN_REQUIRED_VERSION 7.44)
+set(CURL_MIN_REQUIRED_VERSION 7.4)
 find_package(CURL ${CURL_MIN_REQUIRED_VERSION} CONFIG QUIET)
 if(NOT CURL_FOUND)
   find_package(CURL ${CURL_MIN_REQUIRED_VERSION} REQUIRED)
 endif()
 message("Libcurl version ${CURL_VERSION_STRING}")
 
+########################## Libcurl per-version found features. #################################
+#                                                                                              #
+#  Option `CURLSSLOPT_NO_REVOKE` will only be available for customers using version >= 7.44    #
+#                                                                                              #
+#  However, the Azure SDK for C++ can use an older version 7.4 without supporting this option   #
+#                                                                                              #
+#    #     #              #            #           #          #            #                   #
+set(OPTION_NO_REVOKE_REQUIRED_VERSION 7.44)
+if(${CURL_VERSION_STRING} VERSION_GREATER_EQUAL ${OPTION_NO_REVOKE_REQUIRED_VERSION})
+  message("The libcurl version found supports the `CURLSSLOPT_NO_REVOKE` option. Building it as part of the transport adapter options.")
+  add_compile_definitions(BUILD_SSL_OPTION_NO_REVOKE)
```

Comment:
1. Prefix with AZ_, especially when the stuff goes into headers.
2. Even if you prefix with AZ_, when users include the file, they won't have `BUILD_SSL_OPTION_NO_REVOKE` defined when they include the header, so the idea won't work. See if you can get this moved into cpp, or if CURL defines its version as a macro, so you can make an `#ifdef` in the code, not in cmake.